libxrdp/xrdp_rdp.c

```diff
@@ -820,12 +823,30 @@ xrdp_rdp_incoming(struct xrdp_rdp *self)
     self->mcs_channel = self->sec_layer->mcs_layer->userid +
                         MCS_USERCHANNEL_BASE;
     DEBUG(("out xrdp_rdp_incoming mcs channel %d", self->mcs_channel));
-    g_strncpy(self->client_info.client_addr,
-              self->sec_layer->mcs_layer->iso_layer->trans->addr,
+    g_strncpy(self->client_info.client_addr, iso->trans->addr,
               sizeof(self->client_info.client_addr) - 1);
-    g_strncpy(self->client_info.client_port,
-              self->sec_layer->mcs_layer->iso_layer->trans->port,
+    g_strncpy(self->client_info.client_port, iso->trans->port,
               sizeof(self->client_info.client_port) - 1);
+
+    /* log TLS version and cipher when TLS is used */
+    /* TODO: client_addr, client_port is empty when IPv6 enabled */
+
+    if (iso->selectedProtocol > PROTOCOL_RDP)
+    {
+        log_message(LOG_LEVEL_INFO,
+                    "TLS connection established from %s: %s with cipher %s",
```

Comment:
How about replacing the initial "TLS" with the actual protocol name? It could be like "TLSv1.2 connection with cipher AES256-GCM-SHA384 established from 127.0.0.1:5678". How about logging the port? It's rarely useful, but it's just a few bytes in the log.